tests/test_speech_tools.py

```diff
@@ -71,3 +78,23 @@ def test_int(self) -> None:
         val = 1234
 
         assert represent_prop(val) == "<1234>"
+
+    def test_mock(self) -> None:
+        val = mock.Mock()
+
+        assert represent_prop(val) is val
+
+    def test_gtlt_object(self) -> None:
+        val = FancyObj()
+
+        assert represent_prop(val) == "<FancyObj>"
+
+    def test_hamcrest_equal_to(self) -> None:
+        val = equal_to(1)
+
+        assert represent_prop(val) == "<1>"
+
+    def test_hamcrest_instance_of(self) -> None:
+        val = instance_of(int)
+
+        assert represent_prop(val) == "an instance of int"
```

Comment:
I'm not sure that these tests get us much more than the `FancyObj` test just above them. These seem like duplicates to me, duplicates which will break if the maintainers of `pyhamcrest` decide to change their strings.

Reply:
These provide coverage for code paths in `represent_prop` we didn't have.  For example:
`val = equal_to(1)`  `val`here is an instance of `IsEqual`.  We expect that represent_prop will do the right thing, this test just confirms that it does.  

You're not wrong about the fact that if `pyhamcrest` changes the `__str__` method we would break, but that's kind of the point.  We currently expect that _they_ are wrapping those objects in `< >`, in which case we don't want `represent_prop` to add another set.  This test will catch that case if they ever do.  (I think that's a good thing)

Reply:
Why don't we just test that `represent_prop("<1>")` looks like `"<1>"`? I don't think we need to test `hamcrest` here.

Reply:
I think it's wise to utilize the hamcrest objects since they are passed into `represent_prop`.  We technically aren't testing the hamcrest object, but `represent_prop`'s ability to handle them.

Reply:
We are testing the `hamcrest` object, though, because if they change what their output looks like, then our tests will fail. That's a test of their `__str__` output, implicitly.

That's why i'd prefer to test the actual logic we have in place by replacing all these `hamcrest` tests with one test that does
```python
assert represent_prop("<1>") == "<1>"
```

Reply:
I thought about it a different way—you want to make sure our logs look nice _with_ `hamcrest`'s output, so you _do_ want these tests to fail if something changes.

I can get behind that!